host_core/lib/host_core/actors/actor_supervisor.ex

```diff
@@ -38,11 +38,44 @@ defmodule HostCore.Actors.ActorSupervisor do
         Logger.error("Failed to download OCI bytes for #{oci}")
         {:stop, err}
 
-      bytes ->
+      {:ok, bytes} ->
         start_actor(bytes |> IO.iodata_to_binary(), oci)
     end
   end
 
+  def live_update(oci) do
+    with {:ok, bytes} <- HostCore.WasmCloud.Native.get_oci_bytes(oci, false, []),
+         {:ok, new_claims} <-
+           HostCore.WasmCloud.Native.extract_claims(bytes |> IO.iodata_to_binary()),
+         {:ok, old_claims} <- HostCore.Claims.Manager.lookup_claims(new_claims.public_key),
+         :ok <- validate_actor_for_update(old_claims, new_claims) do
```

Comment:
I've never seen this syntax before, is this just a bunch of async calls? Do they happen in sequence?

Reply:
It reads like "once I've done these operations in this order, do the following code", just making sure I understand properly

Reply:
this is synchronous code. The `with` statement basically lets you run a bunch of code in sequence, and if any of the pattern matches fail, then the `else` clause is run. [with](https://www.openmymind.net/Elixirs-With-Statement/)

Reply:
Thanks 🙂